sympy/polys/polyfuncs.py

```diff
@@ -239,17 +239,26 @@ def interpolate(data, x):
 
     """
     n = len(data)
+    poly = None
 
     if isinstance(data, dict):
         X, Y = list(zip(*data.items()))
+        poly = interpolating_poly(n, x, X, Y)
     else:
         if isinstance(data[0], tuple):
             X, Y = list(zip(*data))
+            poly = interpolating_poly(n, x, X, Y)
         else:
-            X = list(range(1, n + 1))
             Y = list(data)
 
-    poly = interpolating_poly(n, x, X, Y)
+            numert = Mul(*[ (x - i) for i in range(1, n + 1) ])
+            denom = Mul(*[ (1 - i) for i in range(2, n + 1) ])
```

Comment:
This looks like a factorial. Could that be used?

BTW, PEP 8 is not always strictly followed in SymPy, but this rule is mostly applied:

    Avoid extraneous whitespace in the following situations:

    Immediately inside parentheses, brackets or braces.

    Yes: spam(ham[1], {eggs: 2})
    No:  spam( ham[ 1 ], { eggs: 2 } )

Reply:
> This looks like a factorial. Could that be used?

Yes, with care for negative sign. Will do.

> BTW, PEP 8 is not always strictly followed in SymPy, but this rule is mostly applied.

I remember seeing this type of formatting used in SymPy. 
Should I remove the spaces before `]` and after `[`?

Reply:
I have also seen that type of formatting but that should probably be avoided in new code. (So please remove the spaces.)

Reply:
> I remember seeing this type of formatting used in SymPy.

In line `138` of `polyfuncs.py`: [here](https://github.com/sidhantnagpal/sympy/blob/lagrange-interpolate/sympy/polys/polyfuncs.py#L138)

Reply:
> I have also seen that type of formatting but that should probably be avoided in new code. (So please remove the spaces.)

Oh, okay. So should I remove the spaces throughout the file (for existing lines as well)?